nixos/modules/services/web-apps/alist.nix

```
@@ -0,0 +1,290 @@
+                default = "";
+                example = "root";
+                description = "database account";
+              };
+              password = lib.mkOption {
+                type = lib.types.oneOf [
+                  lib.types.str
+                  (lib.types.attrsOf lib.types.path)
+                ];
+                default = "";
+                example = lib.literalExpression ''
+                  {
+                    _secret = /run/agenix/alist-db-password;
+                  };
+                '';
+                description = "database password";
+              };
+              name = lib.mkOption {
+                type = lib.types.str;
+                default = "";
+                example = "alist";
+                description = "database name";
+              };
+              db_file = lib.mkOption {
+                type = lib.types.str;
+                default = "/var/lib/alist/data.db";
+                description = "Database location, used by sqlite3";
+              };
+              table_prefix = lib.mkOption {
+                type = lib.types.str;
+                default = "alist_";
+                description = "database table name prefix";
+              };
+              ssl_mode = lib.mkOption {
+                type = lib.types.str;
+                default = "";
+                description = ''
+                  To control the encryption options during the SSL handshake, the parameters can be searched by themselves, or check the answer from ChatGPT on https://alist.nn.ci/config/configuration.html#database
+                '';
+              };
+              dsn = lib.mkOption {
+                type = lib.types.str;
+                default = "";
+                description = ''
+                  Use this option to configure a more flexible database connection such as `unix`
+                  The format needs to match the following:
+                  `[username[:password]@][protocol[(address)]]/dbname[?param1=value1&...&paramN=valueN]`
+                '';
+              };
+            };
+            scheme = {
+              address = lib.mkOption {
+                type = lib.types.str;
+                default = "[::1]";
+                description = ''
+                  The http/https address to listen on, default `[::1]`
+                '';
+              };
+              http_port = lib.mkOption {
+                type = lib.types.oneOf [
+                  (lib.types.enum [ (-1) ])
+                  lib.types.port
+                ];
+                default = 5244;
+                description = ''
+                  The http port to listen on, default `5244`, set it to `-1` to disable `http`
+                '';
+              };
+              https_port = lib.mkOption {
+                type = lib.types.oneOf [
+                  (lib.types.enum [ (-1) ])
+                  lib.types.port
+                ];
+                default = (-1);
```

```suggestion
                default = -1;
```
Useless parentheses. Anyway, I'm not sure that -1 is better than null

---

Using `-1` to disable http and https features is the upstream method. Since the config file is  generated directly from the settings, I think it is best to conform to the upstream.